pilot/proxy/envoy/discovery_test.go

```diff
@@ -629,6 +681,21 @@ func TestListenerDiscoverySidecarAuthOptOut(t *testing.T) {
 	mock.HelloService.Ports[0].AuthenticationPolicy = proxyconfig.AuthenticationPolicy_INHERIT
 }
 
+func TestListenerDiscoverySidecarAuthMigration(t *testing.T) {
+	mesh := makeMeshConfig()
+	registry := memory.Make(model.IstioConfigTypes)
+
+	// Auth migration for port 80 (http, index 0) and port 90 (tcp, index 1)
+	mock.HelloService.Ports[0].AuthMigrationPort = &model.AuthMigrationPort{16080, proxyconfig.AuthenticationPolicy_MUTUAL_TLS, false}
+	mock.HelloService.Ports[2].AuthMigrationPort = &model.AuthMigrationPort{16090, proxyconfig.AuthenticationPolicy_NONE, true}
+	ds := makeDiscoveryService(t, registry, &mesh)
+	url := fmt.Sprintf("/v1/listeners/%s/%s", "istio-proxy", mock.HelloProxyV0.ServiceNode())
+	response := makeDiscoveryRequest(ds, "GET", url, t)
+	compareResponse(response, "testdata/lds-v0-none-auth-migration.json", t)
+	mock.HelloService.Ports[0].AuthMigrationPort = nil
+	mock.HelloService.Ports[2].AuthMigrationPort = nil
+}
```

Comment:
why do you need to test listener discovery?

Reply:
We expect extra listeners are added for migraion ports (16080, 16090)